src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/Process_StreamTests.cs

```diff
@@ -63,25 +64,31 @@ public void Process_SyncOutputStream()
         public void Process_AsyncOutputStream()
         {
             {
+                AutoResetEvent are = new AutoResetEvent(false);
+                int expectedLines = 2;
                 StringBuilder sb = new StringBuilder();
                 Process p = CreateProcessStream();
                 p.StartInfo.RedirectStandardOutput = true;
-                p.OutputDataReceived += (s, e) => sb.Append(e.Data);
+                p.OutputDataReceived += (s, e) => { sb.Append(e.Data); if (--expectedLines == 0) are.Set(); };
```

Comment:
According to the docs, the answer is yes. Although, I've been reading more and it looks like the guidance in this case is to call WaitForExit() after getting true from WaitForExit(timeout). 

"When standard output has been redirected to asynchronous event handlers, it is possible that output processing will not have completed when this method returns. To ensure that asynchronous event handling has been completed, call the WaitForExit() overload that takes no parameter after receiving a true from this overload." 

I'll probably switch to this. If it works as expected, it will be an easier pattern to repeat in the other tests.